superset-frontend/packages/superset-ui-core/src/query/types/Datasource.ts

```diff
@@ -22,6 +22,10 @@ import { Metric } from './Metric';
 export enum DatasourceType {
   Table = 'table',
   Druid = 'druid',
+  Query = 'query',
+  Dataset = 'sl_dataset',
```

Comment:
we're calling this "dataset" on the backend. We may have to change this, but it's fine for now since we're not using it.

Reply:
```suggestion
  Dataset = 'dataset',
```
On second thought, let's just change it now so that we don't forget.